ansible/roles/ocp4-workload-dil-agile-integration/tasks/provision_sso.yaml

```diff
@@ -0,0 +1,57 @@
+---
+# Use the SSO Operator to Install Apicurito
+
+- name: Evaluate {{sso_project }} namespace if not exists 
+  k8s:
+    api_version: v1
+    kind: Namespace
+    name: '{{ sso_project }}'
+    state: present
+
+- name: Evaluate SSO Operator Group
+  k8s:
+    state: present
+    resource_definition: "{{ lookup('template', 'sso-operatorgroup.yaml.j2') }}"
+
+- name: Evaluate SSO Subscription
+  k8s:
+    state: present
+    resource_definition: "{{ lookup('template', 'sso-subscription.yaml.j2') }}"
+
+- name: Wait for SSO operator to install
+  k8s_info:
+    api_version: apiextensions.k8s.io/v1
+    kind: CustomResourceDefinition
+    name: keycloaks.keycloak.org
+  register: crd_sso
+  until: crd_sso.resources | list | length == 1
+  retries: 10
+  delay: 30
+
+- name: Evaluate the SSO Instance CR
+  k8s:
+    state: present
+    resource_definition:  "{{ lookup('template', 'sso-instance.yaml.j2') }}"
+
+- name: Wait for SSO to be running
+  uri:
+    url: https://keycloak-{{ sso_project }}.{{ route_subdomain }}
+    status_code: '200'
+  register: result
+  until: result.status == 200
+  retries: 90
+  delay: 30    
+
+- pause:
+    minutes: 2
+
+- name: Create Threescale realm
+  k8s:
+    state: present
+    resource_definition:  "{{ lookup('template', 'sso-3scale-realm.yaml.j2') }}"
+
+- name: Create User realms
+  include_tasks: user_sso_realm.yaml
+  with_sequence: start=1 end={{ num_users }} 
+  when:
+    - num_users|int > 0
```

Comment:
Switch to `loop` and set a `loop_var` with `loop_control`

Reply:
I recommend setting `loop_var` when calling `include_tasks` in `loop`. It is dangerous not to and can lead to some really surprising errors if the included tasks file has any `loop` blocks if we don't have this.

```
- name: Create User realms
  include_tasks: user_sso_realm.yaml
  loop: "{{ range(1, num_users|int + 1, 1)|list }}"
  loop_control:
    loop_var: __user_number
  when:
    - num_users|int > 0 
```